src/Microsoft.OData.Edm/Parameterized.Microsoft.OData.Edm.cs

```diff
@@ -642,6 +642,20 @@ internal static class Strings {
             return Microsoft.OData.Edm.EntityRes.GetString(Microsoft.OData.Edm.EntityRes.EdmModel_Validator_Semantic_KeyPropertyMustBelongToEntity, p0, p1);
         }
 
+        /// <summary>
+        /// A string like "The 'Edm.PrimitiveType' cannot be used as the type of a key property '{0}' of an entity type '{1}'."
+        /// </summary>
+        internal static string EdmModel_Validator_Semantic_EdmPrimitiveTypeCannotBeUsedAsTypeOfKey(object p0, object p1) {
+            return Microsoft.OData.Edm.EntityRes.GetString(Microsoft.OData.Edm.EntityRes.EdmModel_Validator_Semantic_EdmPrimitiveTypeCannotBeUsedAsTypeOfKey, p0, p1);
+        }
+
+        /// <summary>
+        /// A string like "The 'Edm.PrimitiveType' cannot be used as the underlying type of '{0}' type '{1}'."
+        /// </summary>
+        internal static string EdmModel_Validator_Semantic_EdmPrimitiveTypeCannotBeUsedAsUnderlyingType(object p0, object p1) {

```

Comment:
looks like p0 is always "type definition" only one argument typeDefinition.FullName() is needed?
Please check the chained definitions as well.

Reply:
It can be used as "enumeration type" as well. 
The rule is:
*The 'Edm.PrimitiveType' cannot be used as the underlying type of type definition or enumeration type. *
Owing that the